static/js/biomedical/disease/data_processing_utils.ts

```diff
@@ -150,3 +151,83 @@ export function getDiseaseSymptomAssociation(
   }
   return rawData;
 }
+
+/**
+ * Fetches the chemical compound disease treatment data
+ * @param data
+ * @returns
+ */
+export function getCompoundDiseaseTreatment(
+  data: GraphNodes
+): CompoundDiseaseTreatmentData[] {
+  // checks if the data is empty
+  if (_.isEmpty(data)) {
+    return [];
+  }
+  const rawData: CompoundDiseaseTreatmentData[] = [];
+  // checks for null values
+  if (_.isEmpty(data.nodes) || _.isEmpty(data.nodes[0].neighbors)) {
+    return [];
+  }
+  for (const neighbour of data.nodes[0].neighbors) {
+    if (neighbour.property !== "diseaseID") {
+      continue;
+    }
+    for (const node of neighbour.nodes) {
+      let typeField = null;
+      let nodeVal = null;
+      let compoundID = null;
+      let compoundName = null;
+      let fdaPhase = null;
+      // check for null or non-existent property values
+      if (_.isEmpty(node.neighbors) || _.isEmpty(node.value)) {
+        continue;
+      }
+      nodeVal = node.value;
+      for (const n of node.neighbors) {
+        if (n.property === "typeOf") {
+          if (_.isEmpty(n.nodes) || _.isEmpty(n.nodes[0].value)) {
+            continue;
+          }
+          typeField = n.nodes[0].value;
+        } else if (n.property === "fdaClinicalTrialPhase") {
+          // check for empty list and null gene values
+          if (_.isEmpty(n.nodes) || _.isEmpty(n.nodes[0].value)) {
+            continue;
+          }
+          fdaPhase = n.nodes[0].value;
+        } else if (n.property === "compoundID") {
+          // check if the list is empty or not
+          if (_.isEmpty(n.nodes) || _.isEmpty(n.nodes[0].value)) {
+            continue;
+          }
+          compoundID = n.nodes[0].value;
+          for (const n1 of n.nodes) {
+            if (n1.neighbors === undefined || _.isEmpty(n1.value)) {
```

Comment:
This is similar to our conversation in the previous PR. You're right, n1.value is never used so that check is unnecessary and for the sake of hierarchy, we're checking n1.neighbors so I just replaced that with `_.isEmpty(n1.neighbors)`. Thanks for pointing it out